reference-implementation/lib/readable-stream.js

```diff
@@ -700,8 +700,52 @@ class ReadableStreamDefaultReader {
 
     ReadableStreamReaderGenericRelease(this);
   }
+
+  iterator({ cancel = true } = {}) {
+    const O = this;
+    if (typeof O !== 'object') {
+      throw new TypeError();
+    }
+    if (this._ownerReadableStream === undefined) {
+      throw new TypeError();
+    }
+    const iterator = Object.create(ReadableStreamDefaultReaderAsyncIteratorPrototype);
+    iterator._reader = this;
+    iterator._cancel = Boolean(cancel);
+    return iterator;
+  }
+
+  [Symbol.asyncIterator]() {
+    const O = this;
+    return O.iterator();
+  }
 }
 
+const ReadableStreamDefaultReaderAsyncIteratorPrototype = Object.setPrototypeOf({
+  _reader: undefined,
+  _cancel: undefined,
+  next() {
+    const O = this;
+    if (typeof O !== 'object') {
+      throw new TypeError();
+    }
+    const reader = O._reader;
+    const read = reader.read;
+    return read.call(reader);
+  },
+  return() {
+    const O = this;
+    if (typeof O !== 'object') {
+      throw new TypeError();
+    }
+    if (O._cancel === true) {
+      const reader = O._reader;
+      const cancel = reader.cancel;
+      cancel.call(reader);
```

Comment:
This does not implement [GetMethod](https://tc39.github.io/ecma262/#sec-getmethod) fully. Additionally I'd prefer using `Reflect.apply()`.

Reply:
We should put a full GetMethod implementation in helpers.js.